### PR TITLE
feat: Introduce `kraft pkg show`

### PR DESCRIPTION
**Pull request by mdsahiloss**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [ ] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

Added a subcommand `show` for command `kraft pkg`.

Now any package information can be seen with the help of command `kraft pkg show packName` in the format of YAML|JSON .

### CODE REVIEW

**Review by craciunoiuc — COMMENTED** [review]
Took an initial look and it seems fine

I remember this working when I last saw it

File: manifest/manager.go [hunk 1]
```diff
@@ -499,3 +499,45 @@ func (m *manifestManager) LocalManifestIndex(ctx context.Context) string {
...
+	})
+
+	for _, pack := range packages {
+		if query.Name() == pack.Name() {
+			metadata = pack.Metadata()
+		}
+	}
+
+	if metadata == nil {
+		return nil, fmt.Errorf("package %s information not found", query.Name())
```
Comment (craciunoiuc) [inline manifest/manager.go]:
```suggestion
		return nil, fmt.Errorf("%s package information not found", query.Name())
```

File: packmanager/umbrella.go [hunk 2]
```diff
@@ -295,3 +297,30 @@ func NewUmbrellaManager(ctx context.Context, constructors []func(*UmbrellaManage
...
+				WithField("format", manager.Format()).
+				Debugf("could not run Show(): %v", err)
+			continue
+		}
+
+		metadatas = append(metadatas, metadata)
+	}
+	if len(metadatas) == 0 {
+		return nil, fmt.Errorf(strings.Join(errList, "\n"))
+	}
```
Comment (craciunoiuc) [inline packmanager/umbrella.go]: I would not propagate both errors every time.
If the package is not found, just specify that no information was found about the package:

```suggestion
	if len(metadatas) == 0 {
		return nil, fmt.Error("no information found about requested package")
	}
```

image

File: packmanager/umbrella.go [hunk 3]
```diff
@@ -295,3 +297,30 @@ func NewUmbrellaManager(ctx context.Context, constructors []func(*UmbrellaManage
...
+		log.G(ctx).WithFields(logrus.Fields{
+			"format": manager.Format(),
+		}).Tracef("showing package information")
+
+		metadata, err := manager.Show(ctx, outputFormat, qopts...)
+		if err != nil {
+			errList = append(errList, err.Error())
+			log.G(ctx).
+				WithField("format", manager.Format()).
+				Debugf("could not run Show(): %v", err)
```
Comment (craciunoiuc) [inline packmanager/umbrella.go]:
```suggestion
				Debugf("could not show package: %v", err)
```

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
Hey @MdSahil-oss 

Tried it out a final time and works fine. Some final comments from my side regarding messages passed to the user.

After this we'll merge, I promise 🤞

File: manifest/manager.go (hunk 1 above)
Comment (mdsahiloss) [reply manifest/manager.go]: Done! 👍🏽

File: packmanager/umbrella.go (hunk 2 above)
Comment (mdsahiloss) [reply packmanager/umbrella.go]: Done! 👍🏽

File: packmanager/umbrella.go (hunk 3 above)
Comment (mdsahiloss) [reply packmanager/umbrella.go]: Done! 👍🏽

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request merged**